engine/src/main/java/org/hibernate/search/bridge/impl/ExtendedBridgeProvider.java

```diff
@@ -58,5 +58,13 @@ public FieldBridge provideFieldBridge(BridgeProviderContext bridgeProviderContex
 		 * not.
 		 */
 		boolean isExplicitlyMarkedAsNumeric();
+
+		/**
+		 * Returns the type of the indexed member/property; it works for arrays and collections too.
+		 *
+		 * @see #getReturnType()
+		 * @return the type of the indexed member
+		 */
+		Class<?> getExactReturnType();
```

Comment:
I prefer the `getProperty[Element]Type` approach, I don't like the `getContainerType` one because sometimes it returns null and force me to check.

But I wanted to keep the patch simple without renaming other existing SPI.

Somewhere in the code I think I saw something like `getElementOrContainerType()`, would that be better?